src/rpc/server/scala/serv/src/main/scala/com/ctc/omega_edit/grpc/EditorService.scala

```diff
@@ -457,6 +472,63 @@ class EditorService(implicit val system: ActorSystem) extends Editor {
             SegmentResponse.of(in.sessionId, offset, ByteString.copyFrom(data))
           )
       }
+
+  // server control
+
+  def checkNoSessionsRunning(): Future[Boolean] =
+    (editors ? SessionCount)
+      .mapTo[Int]
+      .map(count =>
+        count compare 0 match {
+          case 0 => true
+          case _ => false
+        }
+      )
+
+  def stopServer(kind: ServerControlKind): Future[ServerControlResponse] =
+    system
+      .terminate()
+      .transform {
+        case Success(_) =>
+          Success(ServerControlResponse(kind, getServerPID(), 0))
+        case Failure(_) =>
+          Success(ServerControlResponse(kind, getServerPID(), 1))
```

Comment:
Swallowing this error seems pretty dubious. Maybe log it and then return whatever response?

Reply:
@arosien What would be the way to log with the logging @scholarsmate had setup for the server?